node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -1,42 +1,87 @@
 package org.bitcoins.node
 
-import akka.Done
 import akka.actor.ActorSystem
 import org.bitcoins.chain.config.ChainAppConfig
 import org.bitcoins.chain.models.BlockHeaderDAO
 import org.bitcoins.core.api.chain.ChainQueryApi.FilterResponse
 import org.bitcoins.core.protocol.BlockStamp
 import org.bitcoins.node.config.NodeAppConfig
 import org.bitcoins.node.models.Peer
+import org.bitcoins.node.networking.P2PClient
+import org.bitcoins.node.networking.peer._
 
-import scala.concurrent.{Future, Promise}
+import scala.concurrent.Future
 
 case class NeutrinoNode(
-    nodePeer: Peer,
+    nodePeers: Vector[Peer],
+    _dataMessageHandler: DataMessageHandler,
     nodeConfig: NodeAppConfig,
     chainConfig: ChainAppConfig,
-    initialSyncDone: Option[Promise[Done]],
     actorSystem: ActorSystem)
     extends Node {
   require(
     nodeConfig.nodeType == NodeType.NeutrinoNode,
     s"We need our Neutrino mode enabled to be able to construct a Neutrino node!")
 
-  implicit override def system: ActorSystem = actorSystem
+  implicit override val system: ActorSystem = actorSystem
 
-  implicit override def nodeAppConfig: NodeAppConfig = nodeConfig
+  implicit override val nodeAppConfig: NodeAppConfig = nodeConfig
 
-  override def chainAppConfig: ChainAppConfig = chainConfig
+  implicit override val chainAppConfig: ChainAppConfig = chainConfig
 
-  override val peer: Peer = nodePeer
+  var peers: Vector[Peer] = nodePeers
+
+  var dataMessageHandler: DataMessageHandler = _dataMessageHandler
+
+  override def updateDataMessageHandler(
+      dataMessageHandler: DataMessageHandler): NeutrinoNode = {
+    this.dataMessageHandler = dataMessageHandler
+    this
+  }
+
+  var clients: Vector[P2PClient] = {
+    peers.map { peer =>
+      val peerMsgRecv: PeerMessageReceiver =
+        PeerMessageReceiver.newReceiver(node = this, peer = peer)
+      P2PClient(context = system,
+                peer = peer,
+                peerMessageReceiver = peerMsgRecv)
+    }
+  }
+
+  var peerMsgSenders: Vector[PeerMessageSender] = {
+    clients.map { client =>
+      PeerMessageSender(client)
+    }
+  }
+
+  def addPeer(peer: Peer): Unit = {
+    if (!peers.contains(peer)) {
+      val peerMsgRecv: PeerMessageReceiver =
+        PeerMessageReceiver.newReceiver(node = this, peer = peer)
+      val client = P2PClient(context = system,
+                             peer = peer,
+                             peerMessageReceiver = peerMsgRecv)
+      val peerMsgSender = PeerMessageSender(client)
+
+      this.peers = peers :+ peer
+      this.clients = clients :+ client
+      this.peerMsgSenders = peerMsgSenders :+ peerMsgSender
+    } else ()
+  }
+
+  def removePeer(peer: Peer): Unit = {
+    this.peers = peers.filter(_ != peer)
+    this.clients = clients.filter(_.peer != peer)
+    this.peerMsgSenders = peerMsgSenders.filter(_.client.peer != peer)
+  }
 
   override def start(): Future[NeutrinoNode] = {
```

Comment:
How do we handle conflicting peers? I.e. if they have different headers or different filter headers? I don't see the compare via binary search and download and ban code.

Also part of the point of having multiple peers should be to speed up processing (though this may be a next PR, an issue would be nice):
```
The header checkpoints allow the client to download filter headers for different intervals from multiple peers in parallel, verifying each range of 1,000 headers against the checkpoints. 
```

Reply:
this was the plan for a follow up PR